### ISSUE TITLE
Composer 2 reports unresolvable packages

### ISSUE THREAD

**Issue by ubuntupunk** [simpleid/simpleid#41]
```
composer create-project simpleid/simpleid:dev-master simpleid --no-dev
Creating a "simpleid/simpleid:dev-master" project at "./simpleid"
Installing simpleid/simpleid (2.0.x-dev 1889737c8adc8e373a604fbf1accd6ebda06bb50)
  - Syncing simpleid/simpleid (dev-master 1889737) into cache
  - Installing simpleid/simpleid (dev-master 1889737): Cloning 1889737c8a from cache
Created project in /var/www/virtual/afro/simpleid
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - wikimedia/composer-merge-plugin[v1.4.0, ..., v1.4.1] require composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - Root composer.json requires wikimedia/composer-merge-plugin ^1.4 -> satisfiable by wikimedia/composer-merge-plugin[v1.4.0, v1.4.1].

You are using Composer 2, which some of your plugins seem to be incompatible with. Make sure you update your plugins or report a plugin-issue to ask them to support Composer 2.
Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.
```

**Comment by ubuntupunk** [comment]
https://github.com/wikimedia/composer-merge-plugin/issues/184

**Comment by kelvinmo** [comment]
Closed by #43
